redis/asyncio/cluster.py

```diff
@@ -232,6 +237,8 @@ def __init__(
         socket_keepalive: bool = False,
         socket_keepalive_options: Optional[Mapping[int, Union[int, bytes]]] = None,
         socket_timeout: Optional[float] = None,
+        retry: Optional["Retry"] = None,
```

Comment:
I'd argue if retry_on_error is added and retry is added - you effectively have duplication. WDYT?